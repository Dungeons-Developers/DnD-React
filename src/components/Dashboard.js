import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper';
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import PersonAddIcon from '@material-ui/icons/PersonAdd';
import AddToQueueIcon from '@material-ui/icons/AddToQueue';
import PeopleIcon from '@material-ui/icons/People';
import Typography from '@material-ui/core/Typography';

function Dashboard({ user }) {

  return (
    <Box
      display="flex"
      flex="1"
      justifyContent="center"
      alignItems="center"
      flexDirection='column'
    >

      <Typography variant="h2">Welcome {user.username}</Typography>
      <Box
        display="flex"
        flex="1"
        justifyContent="center"
        alignItems="center"
      >
        <Box mx=".5rem">
          <Typography align='center' variant='overline' component='p'>Create Character</Typography>
          <Tooltip title='Create Character'>
            <Link to='/create-character'>
              <Paper elevation={3}>
                <Box p="3rem" >
                  <IconButton aria-label="Create Character">
                    <PersonAddIcon style={{ fill: '#BE2224' }} />
                  </IconButton>
                </Box>
              </Paper>
            </Link>
          </Tooltip>
        </Box>
        <Box mx=".5rem">
          <Typography align='center' variant='overline' component='p'>Join Campaign</Typography>
          <Tooltip title='Join Campaign'>
            <Link to='/join'>
              <Paper elevation={3}>
                <Box p="3rem">
                  <IconButton aria-label="Join Campaign">
                    <PeopleIcon style={{ fill: '#BE2224' }} />
                  </IconButton>
                </Box>
              </Paper>
            </Link>
          </Tooltip>
        </Box>
        <Box mx=".5rem">
          <Typography align='center' variant='overline' component='p'>Create Campaign</Typography>
          <Tooltip title='Create Campaign'>
            <Link to='/create-campaign'>
              <Paper elevation={3}>
                <Box p="3rem">

                  <IconButton aria-label="Create Campaign">
                    <AddToQueueIcon style={{ fill: '#BE2224' }} />
                  </IconButton>
                </Box>
              </Paper>
            </Link>
          </Tooltip>
        </Box>
      </Box>
    </Box>
  );
}

const mapStateToProps = (state) => ({
  user: state.users.user
});

export default connect(mapStateToProps)(Dashboard);