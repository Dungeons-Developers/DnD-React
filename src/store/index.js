import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

// import slices here

import users from './slices/user-slice';
import campaign from './slices/campaign-slice';
import characters from './slices/character-slice';
import theme from './slices/theme-slice';

let reducers = combineReducers( { users, campaign, characters, theme } );

let store = configureStore( { reducer: reducers } );

export default store;

