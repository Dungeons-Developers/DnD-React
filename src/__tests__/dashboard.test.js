import React from 'react';
import {mount} from 'enzyme';

import {BrowserRouter} from 'react-router-dom'
import {Provider} from 'react-redux';
import store from '../store';

import Dashboard from '../components/Dashboard';

describe('dashboard tests', () => {

  let component = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>
    </Provider>
  );

  it('matches snapshot?', () => {
    expect(component).toMatchSnapshot();
  });

});