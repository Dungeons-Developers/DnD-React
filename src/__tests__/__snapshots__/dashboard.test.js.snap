// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard tests matches snapshot? 1`] = `ReactWrapper {}`;
